docs/articles/documentation/guides/advanced-guides/authentication.md

```diff
@@ -76,17 +63,9 @@ const admin = Role('http://example.com/login', async t => {
 
 After you create the roles, you can switch between users at any moment except for the role initialization code.
 
-If you switch to a role for the first time in test run, the browser will be navigated from the original page to a login page where the role initialization code will be executed. Then the original page will be reloaded with new credentials (you can use the [preserveUrl](#optionspreserveurl) option to disable redirect to the original page). If you switch to a role that has already been initialized, TestCafe simply reloads the current page with the appropriate credentials.
+If you switch to a role for the first time in test run, the browser will be navigated from the original page to a login page where the role initialization code will be executed. Then the original page will be reloaded with new credentials (you can use the [preserveUrl](../../reference/test-api/role/constructor.md#optionspreserveurl) option to disable redirect to the original page). If you switch to a role that has already been initialized, TestCafe simply reloads the current page with the appropriate credentials.
```

Comment:
Can we say:
If you switch to a role for the first time in a test run, the browser navigates from the original page to a login page where role initialization code is executed. Then, the original page is reloaded with new credentials (you can use the [preserveUrl](../../reference/test-api/role/constructor.md#optionspreserveurl) option to disable the redirect to that page). If you switch to a role that was already initialized, TestCafe simply reloads the current page with corresponding credentials.

Reply:
> Also, for my understanding, can you clarify the meaning of "original page"? Is this the page that you're on when you switch the role?

Yes, that's right.
Your version is fine, thanks!